pkg/controllers/hypershiftdeployment_controller.go

```diff
@@ -108,20 +109,24 @@ func (r *HypershiftDeploymentReconciler) Reconcile(ctx context.Context, req ctrl
 					hypdeployment.MisConfiguredReason)
 		}
 
-		secretName := hyd.Spec.Infrastructure.CloudProvider.Name
-		err = r.Client.Get(r.ctx, types.NamespacedName{Namespace: hyd.Namespace, Name: secretName}, &providerSecret)
+		err = r.Client.Get(r.ctx, types.NamespacedName{Namespace: hyd.Namespace, Name: providerSecretName}, &providerSecret)
 		if err != nil {
 			log.Error(err, "Could not retrieve the provider secret")
 			return ctrl.Result{RequeueAfter: 30 * time.Second, Requeue: true},
 				r.updateStatusConditionsOnChange(&hyd,
 					hypdeployment.ProviderSecretConfigured,
 					metav1.ConditionFalse,
-					"The secret "+secretName+" could not be retreived from namespace "+hyd.Namespace,
+					"The secret "+providerSecretName+" could not be retreived from namespace "+hyd.Namespace,
 					hypdeployment.MisConfiguredReason)
 		}
-		if err := r.updateStatusConditionsOnChange(&hyd, hypdeployment.ProviderSecretConfigured, metav1.ConditionTrue, "Retreived secret "+secretName, string(hypdeployment.AsExpectedReason)); err != nil {
+		if err := r.updateStatusConditionsOnChange(&hyd, hypdeployment.ProviderSecretConfigured, metav1.ConditionTrue, "Retreived secret "+providerSecretName, string(hypdeployment.AsExpectedReason)); err != nil {
 			return ctrl.Result{}, err
 		}
+	} else if providerSecretName != "" {
```

Comment:
We didn't used to get the providerSecret for configure=F, and non-Azure paths.

Now, we get it for all paths, but it's still optional for the other paths (continue after logging a msg)